core-bundle/assets/styles/component/message.pcss

```diff
@@ -0,0 +1,144 @@
+@keyframes toast {
+    0% {
+        grid-template-rows: 0fr;
+        transform: translateY(-2em);
+        opacity: 0;
+        margin-bottom: 0;
+    }
+    10% {
+        grid-template-rows: 1fr;
+    }
+    100% {
+        transform: translateY(0);
+        opacity: 1;
+        margin-bottom: 1em;
+    }
+}
+
+@keyframes untoast {
+    0% {
+        grid-template-rows: 1fr;
+        opacity: 1;
+        margin-bottom: 1em;
+    }
+    100% {
+        grid-template-rows: 0fr;
+        opacity: 0;
+        margin-bottom: 0;
+    }
+}
+
+*[data-controller="contao--message-outlet"] {
+    position: absolute;
+    top: 3.75em;
+    left: 0;
+    right: 0;
+
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    z-index: 110;
+
+    pointer-events: none;
+
+    .message {
+        /* We use grid-template-rows to animate from 0 to auto */
+        display: grid;
+        animation: toast forwards .5s;
+        margin: 0 auto;
+        max-width: 80ch;
+
+        &[aria-hidden="true"] {
+            animation: untoast forwards .5s;
+        }
+
+        .inner {
+            &, * {
+                pointer-events: auto;
+            }
+
+            grid-row: 1 / span 2;
+            overflow: hidden;
+
+            /*max-width: 80ch;*/
+            overflow-wrap: anywhere;
+            line-height: 1.3;
+            box-sizing: border-box;
+            border-radius: 6px;
+            box-shadow: rgba(0, 0, 0, .2) 3px 3px 10px, 0 0 1px var(--border) inset;
+
+            background-repeat: no-repeat;
+            background-position: .8em center;
+            background-color: var(--content-bg);
+
+            button.close {
+                overflow: hidden;
+
+                position: absolute;
+                top: 4px;
+                right: 4px;
+                width: 16px;
+                height: 16px;
+                background: var(--content-border);
+                color: var(--text);
+
+                font-size: .75rem;
+                display: grid;
+                border: none;
+                place-items: center;
+                border-radius: 50%;
+                padding: 0;
+                opacity: .5;
+                font-family: -apple-system, system-ui, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;
+
+                &:hover {
+                    opacity: 1;
+                    background: var(--gray);
+                }
+
+            }
+
+            .content {
+                padding: 12px 24px 12px 32px;
+
+                .token {
+                    background-color: var(--code-bg);
+                    color: var(--text);
+                    padding: 0 4px;
+                }
+            }
+
+
+            position: relative;
```

Comment:
*(Coding Style)* Should be moved to the top I think.

Reply:
It belongs to the part below, but I agree it's cleaner above. Fixed in the last commit.